cluster_vcf_records/vcf_file_read.py

```diff
@@ -1,13 +1,37 @@
+import gzip
 import itertools
 import operator
 import logging
 import multiprocessing
-
-import pyfastaq
+import os
 
 from cluster_vcf_records import vcf_record
 
 
+def open_vcf_file_for_reading(filename):
+    if filename.endswith(".gz"):
+        try:
+            return gzip.open(filename, "rt")
```

Comment:
This will "fail" on a compressed BCF file. I.e the contents will be binary.

Reply:
Would anyone really gzip a bcf file?

Reply:
bcftools view fails on a gzipped bcf file

Reply:
> Would anyone really gzip a bcf file?

That's generally the form I use. But the convention is to just use `.bcf`. I think it technically uses `bgzip` to compress. You can see if the bcf file is invalid with `htsfile in.bcf`

Reply:
Hang on, are you saying you produce the usual binary `.bcf` file, then gzip it to make a `.bcf.gz` file?

Reply:
No, I produce it specifying `-O b` from `bcftools`, which provides compressed BCF

Reply:
> bcftools view fails on a gzipped bcf file

Just thinking also, this might be a problem if this BCF criteria isn't met: *"all info and genotype fields must have their full types specified"*. Or full specification of all contigs.

Reply:
ok, so I agree this is an issue. We could have:
* plain text vcf
* (b)gzipped vcf
* bcf
* compressed (bgzipped) bcf

I think it's safe to assume that `.vcf` means a plain text vcf, and `.vcf.gz` means a (b)gzipped `.vcf` file. I always thought that people used the extension `.bcf` whether not the file has been bgzipped. Like I wouldn't expect a BAM file to have the extension `.bam.gz`.

Reply:
> I always thought that people used the extension .bcf whether not the file has been bgzipped. 

Yeah, they normally do. See https://github.com/brentp/cyvcf2/pull/156 for a discussion of this.